### PR TITLE
Feature/rad form

### PR DESCRIPTION
**Pull request by noahmanger**
This implements a very basic proof of concept for the RAD contact form from #612 . This makes use of Django's built-in [forms API](https://docs.djangoproject.com/en/1.10/ref/forms/) to do all of the required / validation / cleaning / etc. This is my first time using this API, though, so please let me know if I did anything wrong.

contact rad - fec

This includes all of the required (and optional) fields as specified. 

Some notes:

The committee field uses the typeahead pulling from api.open.fec.gov to supply a committee ID to a hidden input:
typeahead

When choosing the "other" category, an optional text field appears:
other

The original issue requested that this certification appear as a popup after submitting the form. That's still possible, but for this first iteration I just included it on the initial screen before the submit button:
image

The design and everything can still be refined. Our goal for this sprint, however, was to get a basic form working so we could validate that everything is connecting as it should.

---

## Connecting to ServiceNow
This introduces a few new env variables to connect to the ServiceNow API. This API is used to generate a list of the categories and it's also what we submit to. @patphongs has the details from ServiceNow for this endpoint.

I was able to connect to the point where I could get the data. When I try to `post`, though, I get a 415 error. I'm emailing our contact there to see what to do about that. 

## Remaining to do:
- [x] Get the form to post to ServiceNow
- [x] Add ServiceNow creds to environments

Hoping that @patphongs can pick up this remaining work with help from other Django folks at 18F.

### CODE REVIEW

**Comment by patphongs** [comment]
@noahmanger Sorry I've been so bogged down with other stuff that I am just looking into this now. I'll let you know what I can find out about the 415 error.

**Comment by noahmanger** [comment]
@patphongs I'm back now so I can keep going with this.

**Comment by noahmanger** [comment]
@patphongs I received verification that this works. There might still be some fine tuning, but I think this portion is good for a technical review if anyone is available.

**Comment by noahmanger** [comment]
Think I got this pretty well polished up.

Form:
submit a question to the reports analysis


Other:
image

Success:
image

**Comment by llienfec** [comment]
Two quick things - 
@emileighoutlaw Under the "Question" box, what would you think about adding some sort of instruction that would trigger the user to add detail to their question? For example, instead of "my report" the report name would be helpful "my Q2 report."  The only things I can come up with are "if in doubt, add detail" or "please be detailed."  Neither are great.

Can we remove the "order in which we received questions" sentence from the submit screen?  You guys have triggered a whole internal discussion about how to prioritize these emails efficiently, and we may not end up on the side of first come, first served. 

...and that's it from me!

Thank you: @emileighoutlaw @jenniferthibault @noahmanger  You added incredible value to this product. We really appreciate the time you took to think through this with us, design it, and help us focus on our users.

**Comment by emileighoutlaw** [comment]
I think it would make sense to add a line that said `Please be as detailed as possible.`, but I wonder if @jenniferthibault has thoughts on where that might fit in best. 

Usually with forms, it's not a great idea to add text inside form fields, because it disappears on the click, and people forget about it. Meanwhile, below the form field, we already have a lot of text with that disclaimer. 


Maybe it would make sense to have it next to the title, like this:

> **Question** (please be as detailed as possible)
> [__________________________________________]

?

**Comment by jenniferthibault** [comment]
A few more tweaks:

- On the success page, all of the text above the success message box should be removed:

screen shot 2017-02-08 at 10 35 39 am

- And the custom subject field label should be `Subject` instead of `Other reason`  
rad-contact-other

- For placing the help text on the Question field, let's use the same placement as we do on the feedback box for suggestions: underneath the label. This differentiates it from the "required" placement

screen shot 2017-02-08 at 10 39 49 am

**Comment by noahmanger** [comment]
Made all those changes.

**Comment by noahmanger** [comment]
This should be all good to go, whenever someone can take a closer look.

**Comment by emileighoutlaw** [comment]
Content looks good on this. So it gets my thumbs up for a merge whenever design/dev approves.

**Comment by xtine** [comment]
I'm trying to view this locally at http://127.0.0.1:8000/registration-and-reporting/question-RAD/, but for some reason I get a 404 error. Did you do any other setup to get it to show?

Commenting before previewing: other than the `{{ form.errors }}` check, is there any other inline error checking of the form? Because we can show a `{% if field.errors %} {{ field.error }} {% endif %}`.

**Comment by xtine** [comment]
I got it working locally by setting `export FEC_FEATURE_RADFORM=True`. I tested this in Chrome, Safari, and Firefox. Browser form error handling (like "Please fill out this field" tooltip) works in Chrome and Firefox, but does not in Safari (10).

Submitting a blank form in Safari gets this error:
screen shot 2017-03-02 at 4 15 54 pm
Which isn't very descriptive, but we can show {{ field.errors }} inline which will show a descriptive error message like "This field is required." or "Enter a valid email address."

**Comment by noahmanger** [comment]
Ooh good catch. I'll update.

**Comment by noahmanger** [comment]
image

**Comment by xtine** [comment]
Inline form errors show up unstyled for me, do you have a pending `fec-style` PR? Otherwise looks great I can merge in now or wait if you want put some tests in this PR as well.

**Comment by noahmanger** [comment]
Alright. Tests added and I added the service now credentials to the dev space. Ready for another set of eyes.

**Comment by xtine** [comment]
@noahmanger: how do you run the form unit tests?

**Comment by noahmanger** [comment]
`fec/manage.py test fec.tests` and `fec/manage.py test home.tests`

File: fec/fec/tests/test_form.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  @mock.patch.object(fec.forms, 'form_categories')
+  @mock.patch.object(fec.forms.requests, 'post')
+  def test_post_to_service_now(self, mock_post, form_categories):
+    form_categories.return_value = [('fake', 'Fake category')]
+    # Test to make sure this method removes the `committee_name` field
+    form = ContactRAD(self.form_data)
+    form.post_to_service_now()
+
+    # call_data = mock_post.call_args[1]['data']
+    # self.assertFalse('committee_name' in call_data)
```
Comment (xtine) [inline fec/fec/tests/test_form.py]: Did you mean to uncomment this?

**Comment by noahmanger** [comment]
Woo! Thanks. We can merge this into `develop` whenever. It'd be good to have it on dev for RAD folks to review.

**Pull request merged**